[question] [workspace] AssertionError: Node Editable

### What is your question?

I am encountering the following error with workspaces. Sorry, I am not able to share repro steps as this involves some proprietary packages. My workspace setup contains an application package and 2 shared libraries. This error is produced as a result of `conan workspace super-install --profile:host default --profile:build default -b=outdated -b=missing`. I am looking for guidance to debug this issue. Thanks!

``` pwsh
...
ERROR: Traceback (most recent call last):
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\cli\cli.py", line 297, in main
    cli.run(args)
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\cli\cli.py", line 193, in run
    command.run(self._conan_api, args[0][1:])
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\cli\command.py", line 199, in run
    sub.run(conan_api, parser, *args)
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\cli\command.py", line 217, in run
    info = self._method(conan_api, parent_parser, self._parser, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\cli\commands\workspace.py", line 228, in workspace_super_install
    conan_api.graph.analyze_binaries(ws_graph, args.build, remotes=remotes, update=args.update,
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\api\subapi\graph.py", line 211, in analyze_binaries
    binaries_analyzer.evaluate_graph(graph, build_mode, lockfile, remotes, update,
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\internal\graph\graph_binaries.py", line 445, in evaluate_graph
    self._evaluate_package_id(node, config_version)
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\internal\graph\graph_binaries.py", line 402, in _evaluate_package_id
    compute_package_id(node, self._modes, config_version, self._hook_manager)
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\internal\graph\compute_pid.py", line 27, in compute_package_id
    req_info = RequirementInfo(dep_node.pref.ref, dep_node.pref.package_id,
                               ^^^^^^^^^^^^^
  File "D:\venv\atlas-async\.venv\Lib\site-packages\conan\internal\graph\graph.py", line 249, in pref
    assert self.ref is not None and self.package_id is not None, "Node %s" % self.recipe
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Node Editable

ERROR: Node Editable
``` 

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

I had to deliberately remove `static-lib-example/[*]` from the workspace to reproduce the problem. `static-lib-example/[*]` (not in the workspace) serves as an intermediate package that depends on an editable package, i.e. `interface-lib-example/[*]`.  To reproduce the error you will have to stage the `static-lib-example` locally. I will try to enumerate the steps more clearly.

ref: https://github.com/AneekCN/conan-cmake-super-build/tree/bug/editable_error 

Here is my conan profile:
```
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=23
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=194
os=Windows
```

Setup:
```
> cd ..\interface-lib-example\
> conan create . 

> cd ..\static-lib-example\
> conan create . 

(optionally, doesn't change anything)
> conan remove interface-lib-example -c

> cd ..\super-build-workspace-example
```

Problem step:
```
❯ conan workspace super-install
WARN: Workspace found: D:\venv\conan2_workspaces\super-build-workspace-example
WARN: Workspace is a dev-only feature, exclusively for testing

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=23
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=194
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=23
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=194
os=Windows


======== Computing dependency graph ========
Graph root
    cli
Requirements
    executable-example/1.0.0 - Editable
    interface-lib-example/1.0.0 - Editable
    plugin-example/1.0.0 - Editable
    shared-lib-example/1.0.0 - Editable
    static-lib-example/1.0.0#93f3fed3b2e98d1fc8765bd7732e8a43 - Cache
Test requirements
    gtest/cci.20210126#7b8ec8c988c27b7f0769ca720054ad9f - Cache
Resolved version ranges
    gtest/[*]: gtest/cci.20210126
    static-lib-example/[*]: static-lib-example/1.0.0
Workspace 'super-build-workspace': Packages build order:
    interface-lib-example/1.0.0
    shared-lib-example/1.0.0
    plugin-example/1.0.0
    executable-example/1.0.0

======== Collapsing workspace packages ========

-------- Collapsed graph --------
Graph root
    conanws.py base project Conanfile: D:\venv\conan2_workspaces\super-build-workspace-example
Requirements
    static-lib-example/1.0.0#93f3fed3b2e98d1fc8765bd7732e8a43 - Cache
Test requirements
    gtest/cci.20210126#7b8ec8c988c27b7f0769ca720054ad9f - Cache

======== Computing necessary packages ========
Computing package_id of gtest/cci.20210126 Cache
Computing package_id of gtest/cci.20210126 Cache
Computing package_id of gtest/cci.20210126 Cache
Computing package_id of gtest/cci.20210126 Cache
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
Connecting to remote 'conancenter' anonymously
gtest/cci.20210126: Main binary package 'aa3ab44c371a678b5269aa80d8c118f02b5a700b' missing
gtest/cci.20210126: Checking 7 compatible configurations
REF:  gtest/cci.20210126
PACKAGE_ID:  a60877c6855fb97c669b0023a2106589bf9d277f
REF:  gtest/cci.20210126
PACKAGE_ID:  a60877c6855fb97c669b0023a2106589bf9d277f
gtest/cci.20210126: Found compatible package 'a60877c6855fb97c669b0023a2106589bf9d277f': compiler.cppstd=14
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
REF:  gtest/cci.20210126
PACKAGE_ID:  aa3ab44c371a678b5269aa80d8c118f02b5a700b
Computing package_id of static-lib-example/1.0.0 Cache
REF:  interface-lib-example/1.0.0
PACKAGE_ID:  None
ERROR: Traceback (most recent call last):
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\cli\cli.py", line 297, in main
    cli.run(args)
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\cli\cli.py", line 193, in run
    command.run(self._conan_api, args[0][1:])
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\cli\command.py", line 199, in run
    sub.run(conan_api, parser, *args)
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\cli\command.py", line 217, in run
    info = self._method(conan_api, parent_parser, self._parser, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\cli\commands\workspace.py", line 228, in workspace_super_install
    conan_api.graph.analyze_binaries(ws_graph, args.build, remotes=remotes, update=args.update,
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\api\subapi\graph.py", line 211, in analyze_binaries
    binaries_analyzer.evaluate_graph(graph, build_mode, lockfile, remotes, update,
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\internal\graph\graph_binaries.py", line 446, in evaluate_graph
    self._evaluate_package_id(node, config_version)
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\internal\graph\graph_binaries.py", line 402, in _evaluate_package_id
    compute_package_id(node, self._modes, config_version, self._hook_manager)
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\internal\graph\compute_pid.py", line 27, in compute_package_id
    req_info = RequirementInfo(dep_node.pref.ref, dep_node.pref.package_id,
                               ^^^^^^^^^^^^^
  File "D:\venv\conan2_workspaces\.venv\Lib\site-packages\conan\internal\graph\graph.py", line 251, in pref
    assert self.ref is not None and self.package_id is not None, "Node %s" % self.recipe
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Node Editable

ERROR: Node Editable
```

--- a/conan/api/subapi/workspace.py
+++ b/conan/api/subapi/workspace.py
@@ -140,9 +140,10 @@ class WorkspaceAPI:
                 )
         return packages
 
-    def open(self, require, remotes, cwd=None):
+    def open(self, ref, remotes, cwd=None):
+        cwd = cwd or os.getcwd()
         app = ConanApp(self._conan_api)
-        ref = RecipeReference.loads(require)
+        ref = RecipeReference.loads(ref) if isinstance(ref, str) else ref
         recipe = app.proxy.get_recipe(ref, remotes, update=False, check_update=False)
 
         layout, recipe_status, remote = recipe
@@ -153,7 +154,7 @@ class WorkspaceAPI:
         conanfile, module = app.loader.load_basic_module(conanfile_path, remotes=remotes)
 
         scm = conanfile.conan_data.get("scm") if conanfile.conan_data else None
-        dst_path = os.path.join(cwd or os.getcwd(), ref.name)
+        dst_path = os.path.join(cwd, ref.name)
         if scm is None:
             conanfile.output.warning("conandata doesn't contain 'scm' information\n"
                                      "doing a local copy!!!")
@@ -202,6 +203,34 @@ class WorkspaceAPI:
         # Check the conanfile is there, and name/version matches
         self._ws.add(ref, full_path, output_folder)
         return ref
+
+    def complete(self, profile_host, profile_build, lockfile, remotes, update):
+        packages = self.packages()
+        for ref, info in packages.items():
+            ConanOutput().title(f"Computing the dependency graph for package: {ref}")
+            gapi = self._conan_api.graph
+            deps_graph = gapi.load_graph_requires([ref], None, profile_host, profile_build,
+                                                  lockfile, remotes, update)
+            deps_graph.report_graph_error()
+            print_graph_basic(deps_graph)
+            for node in deps_graph.nodes[1:]:  # Exclude the current virtual root
+                if node.recipe != RECIPE_EDITABLE:
+                    # sanity check, a pacakge in the cache cannot have dependencies to the workspace
+                    deps_edit = [d.node for d in node.transitive_deps.values()
+                                 if d.node.recipe == RECIPE_EDITABLE]
+                    if deps_edit:
+                        full_path = os.path.join(self._folder, node.name, "conanfile.py")
+                        dep_ref = node.ref
+                        ConanOutput().info(f"Adding to workspace {dep_ref}")
+                        try:
+                            self._ws.add(dep_ref, full_path, output_folder=None)
+                        except ConanException:
+                            if os.path.isfile(full_path):
+                                raise
+                            ConanOutput().info(f"Conanfile in {node.name} not found, trying "
+                                               "to open it first")
+                            self.open(dep_ref, remotes, cwd=self._folder)
+                            self._ws.add(dep_ref, full_path, output_folder=None)
 
     @staticmethod
     def init(path):
@@ -305,15 +334,10 @@ class WorkspaceAPI:
         result = DepsGraph()  # TODO: We might need to copy more information from the original graph
         result.add_node(root)
         conanfile.workspace_packages = {}
+
+        self._check_graph(deps_graph)
         for node in deps_graph.nodes[1:]:  # Exclude the current root
             if node.recipe != RECIPE_EDITABLE:
-                # sanity check, a pacakge in the cache cannot have dependencies to the workspace
-                deps_edit = [d.node for d in node.transitive_deps.values()
-                             if d.node.recipe == RECIPE_EDITABLE]
-                if deps_edit:
-                    raise ConanException(f"Workspace definition error. Package {node} in the "
-                                         f"Conan cache has dependencies to packages "
-                                         f"in the workspace: {deps_edit}")
                 result.add_node(node)
                 continue
             # At the moment we are exposing the full conanfile, docs will warn against usage of
@@ -335,6 +359,20 @@ class WorkspaceAPI:
             result.add_edge(root, t.node, r)
 
         return result
+
+    @staticmethod
+    def _check_graph(graph):
+        for node in graph.nodes[1:]:  # Exclude the current root
+            if node.recipe != RECIPE_EDITABLE:
+                # sanity check, a pacakge in the cache cannot have dependencies to the workspace
+                deps_edit = [d.node for d in node.transitive_deps.values()
+                             if d.node.recipe == RECIPE_EDITABLE]
+                if deps_edit:
+                    raise ConanException(f"Workspace definition error. Package {node} in the "
+                                         f"Conan cache has dependencies to packages "
+                                         f"in the workspace: {deps_edit}\n"
+                                         "Try the 'conan workspace complete' to open/add "
+                                         "intermediate packages")
 
     def export(self, lockfile=None, remotes=None):
         self._check_ws()
@@ -376,6 +414,9 @@ class WorkspaceAPI:
                                                              remotes, update)
             deps_graph.report_graph_error()
             print_graph_basic(deps_graph)
+
+            self._check_graph(deps_graph)
+
             conan_api.graph.analyze_binaries(deps_graph, build_mode, remotes=remotes, update=update,
                                              lockfile=lockfile)
             print_graph_packages(deps_graph)

--- a/conan/api/subapi/workspace.py
+++ b/conan/api/subapi/workspace.py
@@ -206,6 +206,10 @@ class WorkspaceAPI:
 
     def complete(self, profile_host, profile_build, lockfile, remotes, update):
         packages = self.packages()
+        if not packages:
+            ConanOutput().info("There are no packages in this workspace, nothing to complete")
+            return
+
         for ref, info in packages.items():
             ConanOutput().title(f"Computing the dependency graph for package: {ref}")
             gapi = self._conan_api.graph
@@ -213,24 +217,31 @@ class WorkspaceAPI:
                                                   lockfile, remotes, update)
             deps_graph.report_graph_error()
             print_graph_basic(deps_graph)
+
+            nodes_to_complete = []
             for node in deps_graph.nodes[1:]:  # Exclude the current virtual root
                 if node.recipe != RECIPE_EDITABLE:
                     # sanity check, a pacakge in the cache cannot have dependencies to the workspace
-                    deps_edit = [d.node for d in node.transitive_deps.values()
-                                 if d.node.recipe == RECIPE_EDITABLE]
-                    if deps_edit:
-                        full_path = os.path.join(self._folder, node.name, "conanfile.py")
-                        dep_ref = node.ref
-                        ConanOutput().info(f"Adding to workspace {dep_ref}")
-                        try:
-                            self._ws.add(dep_ref, full_path, output_folder=None)
-                        except ConanException:
-                            if os.path.isfile(full_path):
-                                raise
-                            ConanOutput().info(f"Conanfile in {node.name} not found, trying "
-                                               "to open it first")
-                            self.open(dep_ref, remotes, cwd=self._folder)
-                            self._ws.add(dep_ref, full_path, output_folder=None)
+                    if any(d.node.recipe == RECIPE_EDITABLE for d in node.transitive_deps.values()):
+                        nodes_to_complete.append(node)
+
+            if not nodes_to_complete:
+                ConanOutput().info("There are no intermediate packages to add to the workspace")
+                return
+
+            for node in nodes_to_complete:
+                full_path = os.path.join(self._folder, node.name, "conanfile.py")
+                dep_ref = node.ref
+                ConanOutput().info(f"Adding to workspace {dep_ref}")
+                try:
+                    self._ws.add(dep_ref, full_path, output_folder=None)
+                except ConanException:
+                    if os.path.isfile(full_path):
+                        raise
+                    ConanOutput().info(f"Conanfile in {node.name} not found, trying "
+                                       "to open it first")
+                    self.open(dep_ref, remotes, cwd=self._folder)
+                    self._ws.add(dep_ref, full_path, output_folder=None)
 
     @staticmethod
     def init(path):

--- a/conan/cli/commands/workspace.py
+++ b/conan/cli/commands/workspace.py
@@ -76,6 +76,29 @@ def workspace_add(conan_api: ConanAPI, parser, subparser, *args):
 
 
 @conan_subcommand()
+def workspace_complete(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Complete the workspace, opening or adding intermediate packages to it that have
+    requirements to other packages in the workspace.
+    """
+    add_common_install_arguments(subparser)
+    add_lockfile_args(subparser)
+    args = parser.parse_args(*args)
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    # The lockfile by default if not defined will be read from the root workspace folder
+    ws_folder = conan_api.workspace.folder()
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile, conanfile_path=ws_folder,
+                                               cwd=None, partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    print_profiles(profile_host, profile_build)
+
+    ConanOutput().box("Workspace completing intermediate packages")
+    conan_api.workspace.complete(profile_host, profile_build, lockfile, remotes, update=args.update)
+
+
+@conan_subcommand()
 def workspace_remove(conan_api: ConanAPI, parser, subparser, *args):
     """
     Remove packages from the current workspace
@@ -200,12 +223,13 @@ def workspace_super_install(conan_api: ConanAPI, parser, subparser, *args):
                            help='The root output folder for generated and build files')
     subparser.add_argument("-d", "--deployer", action="append",
                            help="Deploy using the provided deployer to the output folder. "
-                             "Built-in deployers: 'full_deploy', 'direct_deploy', 'runtime_deploy'")
+                                "Built-in deployers: 'full_deploy', 'direct_deploy', "
+                                "'runtime_deploy'")
     subparser.add_argument("--deployer-folder",
                            help="Deployer output folder, base build folder by default if not set")
     subparser.add_argument("--deployer-package", action="append",
                            help="Execute the deploy() method of the packages matching "
-                             "the provided patterns")    
+                                "the provided patterns")
     subparser.add_argument("--envs-generation", default=None, choices=["false"],
                            help="Generation strategy for virtual environment files for the root")
     add_common_install_arguments(subparser)

--- a/conan/internal/model/workspace.py
+++ b/conan/internal/model/workspace.py
@@ -57,7 +57,8 @@ class Workspace:
         return data or {}
 
     def add(self, ref, path, output_folder):
-        assert os.path.isfile(path)
+        if not path or not os.path.isfile(path):
+            raise ConanException(f"Cannot add to workspace. File not found: {path}")
         path = self._conan_rel_path(os.path.dirname(path))
         editable = {
             "path": path,

--- a/test/integration/workspace/test_workspace.py
+++ b/test/integration/workspace/test_workspace.py
@@ -433,10 +433,21 @@ class TestComplete:
                 "pkgd/conanfile.py": GenConanfile("pkgd", "0.1").with_requires("pkgb/0.1",
                                                                                "pkgc/0.1")})
 
+        c.run("workspace complete")  # Does nothing, but it doesn't fail
+        assert "There are no packages in this workspace, nothing to complete" in c.out
+        c.run("workspace info")
+        assert "packages: (empty)" in c.out
+
         for pkg in ("pkga", "pkgx", "pkgb", "pkgc", "pkgd"):
             c.run(f"export {pkg}")
 
         c.run("workspace add pkgd")
+        c.run("workspace complete")  # Does nothing, but it doesn't fail
+        assert "There are no intermediate packages to add to the workspace" in c.out
+        c.run("workspace info")
+        assert "pkgd/0.1" in c.out
+        assert "pkga/0.1" not in c.out
+
         c.run("workspace add pkga")
         c.run("workspace install --build=missing", assert_error=True)
         assert "Workspace definition error. Package pkgb/0.1 in the Conan cache" in c.out

--- a/test/integration/workspace/test_workspace.py
+++ b/test/integration/workspace/test_workspace.py
@@ -421,6 +421,66 @@ class TestOpenAdd:
         assert "pkgb/0.1: WARN: BUILD PKGB!" in c.out
 
 
+class TestComplete:
+    def test_complete_add(self):
+        c = TestClient(light=True)
+        c.save({"conanws.yml": "",
+                "pkga/conanfile.py": GenConanfile("pkga", "0.1"),
+                "pkgx/conanfile.py": GenConanfile("pkgx", "0.1"),
+                "pkgb/conanfile.py": GenConanfile("pkgb", "0.1").with_requires("pkga/0.1"),
+                "pkgc/conanfile.py": GenConanfile("pkgc", "0.1").with_requires("pkga/0.1",
+                                                                               "pkgx/0.1"),
+                "pkgd/conanfile.py": GenConanfile("pkgd", "0.1").with_requires("pkgb/0.1",
+                                                                               "pkgc/0.1")})
+
+        for pkg in ("pkga", "pkgx", "pkgb", "pkgc", "pkgd"):
+            c.run(f"export {pkg}")
+
+        c.run("workspace add pkgd")
+        c.run("workspace add pkga")
+        c.run("workspace install --build=missing", assert_error=True)
+        assert "Workspace definition error. Package pkgb/0.1 in the Conan cache" in c.out
+        c.run("workspace info")
+        assert "pkgb/0.1" not in c.out
+        assert "pkgc/0.1" not in c.out
+
+        c.run("workspace complete")
+        c.run("workspace info")
+        assert "pkgb/0.1" in c.out
+        assert "pkgc/0.1" in c.out
+        assert "pkgx/0.1" not in c.out
+
+    def test_complete_open(self):
+        c = TestClient(light=True)
+        c.save({"pkgx/conanfile.py": GenConanfile("pkgx", "0.1"),
+                "pkgb/conanfile.py": GenConanfile("pkgb", "0.1").with_requires("pkga/0.1"),
+                "pkgc/conanfile.py": GenConanfile("pkgc", "0.1").with_requires("pkga/0.1",
+                                                                               "pkgx/0.1")})
+
+        for pkg in ("pkgx", "pkgb", "pkgc"):
+            c.run(f"export {pkg}")
+
+        c.save({"conanws.yml": "",
+                "pkga/conanfile.py": GenConanfile("pkga", "0.1"),
+                "pkgd/conanfile.py": GenConanfile("pkgd", "0.1").with_requires("pkgb/0.1",
+                                                                               "pkgc/0.1")},
+               clean_first=True)
+
+        c.run("workspace add pkgd")
+        c.run("workspace add pkga")
+        c.run("workspace install --build=missing", assert_error=True)
+        assert "Workspace definition error. Package pkgb/0.1 in the Conan cache" in c.out
+        c.run("workspace info")
+        assert "pkgb/0.1" not in c.out
+        assert "pkgc/0.1" not in c.out
+
+        c.run("workspace complete")
+        c.run("workspace info")
+        assert "pkgb/0.1" in c.out
+        assert "pkgc/0.1" in c.out
+        assert "pkgx/0.1" not in c.out
+
+
 class TestWorkspaceBuild:
 
     def test_build(self):

